fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/DepositAccountAssembler.java

```diff
@@ -341,7 +342,7 @@ public SavingsAccount assembleFrom(final JsonCommand command, final AppUser subm
                     prodRecurringDetail.recurringDetail());
 
             RecurringDepositAccount rdAccount = RecurringDepositAccount.createNewApplicationForSubmittal(client, group, product,
-                    fieldOfficer, accountNo, externalId, accountType, submittedOnDate, submittedBy, interestRate,
+                    fieldOfficer, accountNo, ExternalIdFactory.produce(externalId), accountType, submittedOnDate, submittedBy, interestRate,
```

Comment:
Here we should generate one if nothing was provided: `ExternalIdFactory.create(...)`